flang/include/flang/Optimizer/Dialect/CUF/CUFOps.td

```diff
@@ -260,9 +263,28 @@ def cuf_KernelOp : cuf_Op<"kernel", [AttrSizedOperandSegments,
         ( `,` `stream` `=` $stream^ )? `>` `>` `>`
         custom<CUFKernelLoopControl>($region, $lowerbound, type($lowerbound),
             $upperbound, type($upperbound), $step, type($step))
+        ( `reduce` `(` $reduceOperands^ `:` type($reduceOperands) `)` )?
+        ( `reduce_attrs` `(` $reduceAttrs^ `)` )?
```

Comment:
Can we move this before the region? We don't have any operation that have operands printed after the region so it would feel a bit werid.

Reply:
I think I found the nicest format. I moved it before region and removed the `reduce_attrs` keyword:
```
    cuf.kernel<<<*, *>>> reduce(%1, %12 : !fir.ref<i32>, !fir.ref<i32> : [#fir.reduce_attr<add>, #fir.reduce_attr<multiply>]) (%arg0 : index) = (%c1 : index) to (%c4 : index)  step (%c1 : index) {
```